### DISCUSSION TITLE
Where should I make API calls for user-edited data (GET, POST, PUT & DELETE calls)?

### DISCUSSION THREAD

**Discussion by crevulus** [vercel/next.js#20196] [Help]
Hi all,

In the Next.js projects I've made so far, I've been doing GET calls in `getStaticProps`. But I can see some obstacles for this method if I want users to be updating their profiles etc.

Let's say a user has already entered their profile info. Then a few days later they come back to edit it. I can retrieve the data in `getStaticProps` no problem, but then if I want the user to submit a form to update that info where can I make that call? the page has already rendered so all of the SSR methods aren't an option.

Is it just the case that I should be using the SSR methods for initial load and then do everything else in React lifecycle methods? Previously I was thinking of the SSR methods as replacements for lifecycle methods, but is this the wrong way to view NEXT's functionality?

**Comment by revskill10** [comment]
I think private data (like profile data) should not be SSG. You should use `useEffect` instead.

**Reply by crevulus** [reply]
I tend to use class components so I guess I'd use `componentDidMount` instead of `useEffect` - is that ill-advised? If I'm using Next should I be making functional components?

**Reply by stigkj** [reply]
As far as I know, using class components is perfectly valid in a Next.js context. <-- @crevulus
